Refactor `@xstate/react` to useSyncExternalStore (#2939)
statelyai/xstate: .changeset/hungry-dryers-bow.md

---
'xstate': patch
---

> Fixed issues with not disposing some cached internal values when stopping interpreters. This could lead to issues when starting such an interpreter again.

Reviewer: ```suggestion
Fixed issues with not disposing some cached internal values when stopping interpreters, which could have led to issues when starting such an interpreter again.
```

Rephrased to clarify that this change _fixes_ the potential issues, instead of introduces them 😅
